pom.xml

```diff
@@ -12,9 +12,9 @@
 
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-        <java.version>17</java.version>
-        <maven.compiler.source>17</maven.compiler.source>
-        <maven.compiler.target>17</maven.compiler.target>
+        <java.version>21</java.version>
+        <maven.compiler.source>21</maven.compiler.source>
+        <maven.compiler.target>21</maven.compiler.target>
```

Comment:
Is it possible to separate this part (switch to java 21) to its own PR that can go first?

The docker image files would need to change at the very least, and I think I caught in a different PR/Issue discussion about other code that also needs fixing.

Thanks!

Reply:
I started this in #410.

Let's get that merged first and then we can merge yours.

Reply:
Did that.